### PR TITLE
feat: add `debug_getRawHeader` method

### LINKED ISSUE #4891: Add `debug_getRawHeader` method
### Problem

Currently, we're missing some `debug_*` methods.

### Solution

- Implement `debug_getRawHeader` method - we should throw `UNSUPPORTED` (TBD) due ot Hedera narrative.
- Check conformity tests
-Add tests (if needed)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
### Description

Currently, we're missing the `debug_getRawHeader` method.

### Solution

- Implement `debug_getRawHeader` method
- Check conformity tests
- Edit **openrpc.json** and **rpc-api.md**
- Add tests (if needed)

### Missing fields in the Hedera concept:

**fee_recipient**  
- Address that receives transaction priority fees and MEV for the block  
- Set by the block proposer
- Separates who proposes the block from who gets paid

**prev_randao**  
- On-chain randomness value from the Proof-of-Stake RANDAO  
- Used for validator selection and committee assignments
- Represents accumulated randomness up to the previous step (not perfectly random)

**withdrawal_root**  
- Merkle root committing to all validator withdrawals in the block  
- Links consensus-layer withdrawals to execution-layer processing

RPC method:
```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "debug_getRawHeader",
  "params": [
    "latest" // block number or block hash
  ]
}
```
Response:
```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": "0xf90211a0..." // RLP encoded header
}
```

Note:
- fee_recipient - use 0.0.801 based on https://hips.hedera.com/hip/hip-1259
- prevRandao - use block.difficulty - `0x0`
- withdrawal_root - use computed hash of an empty Trie object - `0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421`

### Related issue(s)

Fixes #4891 

### Testing Guide

1.
2.
3.

### Changes from original design (optional)

N/A

### Additional work needed (optional)

N/A

### Checklist

- [ ] I've assigned an assignee to this PR and related issue(s) (if applicable)
- [ ] I've assigned a label to this PR and related issue(s) (if applicable)
- [ ] I've assigned a milestone to this PR and related issue(s) (if applicable)
- [ ] I've updated documentation (code comments, README, etc. if applicable)
- [ ] I've done sufficient testing (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/debug.ts [hunk 1]
```diff
@@ -136,6 +136,41 @@ export class DebugImpl implements Debug {
...
+    const block: Block | null =
+      blockNrOrHash.length === 66
+        ? await this.eth.getBlockByHash(blockNrOrHash, false, requestDetails)
+        : await this.eth.getBlockByNumber(blockNrOrHash, false, requestDetails);
+
+    if (!block) {
+      return constants.EMPTY_HEX;
+    }
+
+    return constants.EMPTY_HEX + Buffer.from(BlockFactory.rlpEncode(block, true)).toString('hex');
```
Comment (jasuwienas) [inline packages/relay/src/lib/debug.ts]: WOuld it make sense to use formatter::toHexString && formatters::prepend0x, here? @natanasow

**Review by jasuwienas — APPROVED** [review]
lgtm...


It’s unfortunate that we have to fetch all the transactions in a block just to return the header. However, as long as we’re responsible for calculating the root, I don’t think there’s any way around it.

Just few nits to consider (non-blockers I guess)

EDIT: but you'llhave to resolve the conflicts first.

**Review by jasuwienas — APPROVED** [review]

**Review by simzzz — APPROVED** [review]

**Review by simzzz — DISMISSED** [review]

File: packages/relay/src/lib/factories/blockFactory.ts [hunk 2]
```diff
@@ -138,10 +138,12 @@ export class BlockFactory {
...
    *
    * @param { Block } block - The block object from eth_getBlockByNumber/Hash
+   * @param {boolean} headerOnly - A flag that determines whether to encode the entire block or only the block header
+   *
    * @returns {Uint8Array} - RLP encoded block as Uint8 array
    */
-  static rlpEncode(block: Block): Uint8Array {
-    if (typeof block.transactions[0] === 'string') {
+  static rlpEncode(block: Block, headerOnly: boolean = false): Uint8Array {
+    if (!headerOnly && typeof block.transactions[0] === 'string') {
```
Comment (jasuwienas) [inline packages/relay/src/lib/factories/blockFactory.ts]: It's not a blocker for me, but I think it would look a little bit cleaner with separate methods, and we wouldn’t need conditional validation logic here.

```ts
encodeHeader(block): RLP => rlp(header(block));
encode(block): RLP => validate(block) && rlp(...header(block), ...payload(block));
```

also because of this the jsdocs are not quite valid any longer (especiall ydescription of the returned value)
```
RLP encode a block based on Ethereum Yellow Paper.
   * @returns {Uint8Array} - RLP encoded block as Uint8 array
```
it should be more like
```
RLP encode a block (or its header only) based on Ethereum Yellow Paper.
   * @returns {Uint8Array} - RLP encoded block (or block header) as Uint8 array
```

**Review by jasuwienas — DISMISSED** [review]
lgtm

File: packages/relay/src/lib/factories/blockFactory.ts [hunk 3]
```diff
@@ -138,10 +138,12 @@ export class BlockFactory {
    * RLP encode a block based on Ethereum Yellow Paper.
    *
    * @param { Block } block - The block object from eth_getBlockByNumber/Hash
+   * @param {boolean} headerOnly - A flag that determines whether to encode the entire block or only the block header
+   *
    * @returns {Uint8Array} - RLP encoded block as Uint8 array
    */
-  static rlpEncode(block: Block): Uint8Array {
-    if (typeof block.transactions[0] === 'string') {
+  static rlpEncode(block: Block, headerOnly: boolean = false): Uint8Array {
```
Comment (BartoszSolkaBD) [inline packages/relay/src/lib/factories/blockFactory.ts]: maybe we should split preparing canonical representation and encoding into two separate steps? I think it would allow us to get rid of `headerOnly` flag, swapping it for `rlpEncodeBlock(block: Block)` and `rlpEncodeBlockHeader(header: BlockHeader)`

Comment (natanasow) [reply packages/relay/src/lib/factories/blockFactory.ts]: Fixed.

File: packages/relay/tests/lib/factories/blockFactory.spec.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        ],
+      };
+
+      const encoded = BlockFactory.rlpEncodeTx(tx);
+      expect(encoded.startsWith('0x04')).to.be.true;
+      expect(encoded.length).to.be.greaterThan(10);
+    });
+  });
+
+  describe('rlpEncode', () => {
```
Comment (quiet-node) [inline packages/relay/tests/lib/factories/blockFactory.spec.ts]: Should we add tests for rlpEndcodeBlock with empty transactions array?
